### ISSUE TITLE
option to include the Schema Name everywhere

### ISSUE THREAD

**Issue by aisbergde** [schemaspy/schemaspy#683]
## Expected Behavior

looking at the documentation it should be possible to understand to which schema the documentation belongs to. This should be clear on each row because it should also be possible to take screenshots or to copy and past parts of the documentation

## Current Behavior

Only table name or view name are displayed, no information about the schema is available

## Possible Solution

option / parameter to enforce schema name on all objects

## Context

Very often my tables and views have the same name and it is very important to show the schema everywhere. An example how I normally work:

- import.Customer
- staging.Customer
- check.Customer
- error.Customer
- anonymization.Customer
- historization.Customer
- integeration.Customer
- DataMartPreparation.Customer
- Datamart01.Customer
- Datamart02.Customer

That's why I need the Schema everywhere, in each row.

**Comment by npetzall** [comment]
Option could be -fqn and it should probably include catalog.schema.table

**Comment by aisbergde** [comment]
I added the option the properties file
`schemaspy.fqn`
but there is no effect. No schemas in the output.

But I can't find this option in the documentation: 
https://schemaspy.readthedocs.io/en/latest/configuration/commandline.html
Are all options documented or are there some hidden options?
How to know them?
I think it would be a good idea to document all options :-)

**Comment by npetzall** [comment]
Sorry it was comment for when/if implemented.

**Comment by aisbergde** [comment]
>Option could be -fqn and it should probably include catalog.schema.table

Catalog would blow up the documentation. I don't want to see the catalog when I document only one database. 

Then we will need different options
- one option to include schema everywhere (this is a must have)
- one option to include also the database (catalog)

The only situation when I would like to see the database would be when I mix databases. In this case I want to be able to set one database as current database and to omit catalog name, but to show database name for objects not in the current database

And I want to be able to define alias for other database to avoid to long names.

**Comment by aisbergde** [comment]
The evaluation of schemaspy was great, but for a usage in a production environment the schema name is required in reports.

I think there should be 3 options controled by this parameter:

- catalog.schema.table
- schema.table
- table

(not only for tables, but for all objects)

Is there something I could contribute to get this feature implemented?
Could this be included into the next milestone?

**Comment by maxrausch** [comment]
Has there been any updates here?

**Comment by npetzall** [comment]
@maxrausch no, no work has been done yet. The report is per schema. So there is context, but since the request is to be able to extract (screenshot) parts of the report the context is lost, since identifieras don't include hierarchy (Catalog and/or Schema depending on usage)

**Comment by maxrausch** [comment]
Thanks for letting me know. Do you ever do paid development to add features?

**Comment by npetzall** [comment]
@maxrausch it has never been discussed. So we haven't done it. I'm not even sure how it could be done.

@jesperolsson-se do you have any idea about how it could be done?

**Comment by jesperolsson-se** [comment]
I think it would be wonderful if we could offer this! Thanks for your suggestion @maxrausch. We have quite the backlog and it isn't always easy to decide The Next Most Important Thing. This could be a helpful metric.

As stated, the project currently doesn't receive any donations and few people vote (with :+1:'s). So I'm unsure of how many people would use paid development so the best would probably be some bare-bones solution. For example - assuming we set up GitHub sponsors or Patreon - people could comment how much they'd pay and we'll trust that the pledge is honored.
